src/Pairing/CyclicGroup.hs

```diff
@@ -2,18 +2,26 @@ module Pairing.CyclicGroup
   ( AsInteger(..)
   , CyclicGroup(..)
   , sumG
+  , FromX(..)
+  , Validate (..)
   ) where
 
 import Protolude
+import Crypto.Random (MonadRandom)
+import Data.ByteString.Builder
+import Data.ByteString as BS
 
 class AsInteger a where
   asInteger :: a -> Integer
 
+type LargestY = Bool
+
 class Monoid g => CyclicGroup g where
   generator :: g
   order :: Proxy g -> Integer
   expn :: AsInteger e => g -> e -> g
   inverse :: g -> g
+  random :: (MonadRandom m) => g -> m g
```

Comment:
This perhaps seems kind of odd to have in a `CyclicGroup` typeclass.

Reply:
I had a version that had a Rand type class but I removed it. My reasoning was that it would be convenient to get a random value for a cyclic group as a standardised interface across G1, G2 and GT. 

I can put the type class back if that make more sense - Fq, Fq2 and Fq12 can be made instances of a Rand as well.